### DISCUSSION TITLE
Should Knowbot be like a bash terminal?

### DISCUSSION THREAD

**Discussion by blairmunroakusa** [interlock-network/knowbot#20] [General]
I am wondering if knowbot should use commands that are "more intuitive" to the average non-dev user.

My original interest was to emulate a bash terminal from discord, intention being _to appeal directly to the dev community_.

My wonder is if this is too niche, and that other orgs would prefer something that isn't so 'techie'. IE, instead of 'grep', use 'search' etc. This makes a difference for more complicated features such as "list all pipe to search".

What do you think? @justerhan @jmercouris @galactic-beyond @DecentralizedDan @rickdeaconx @Treevuhh 
I would appreciate the feedback.

**Comment by galactic-beyond** [comment]
So my inclination is to imitate `bash`. The idea being that it could help us attract developers, and also it might intrigue normal users (i.e. what are these cryptic commands?).

But that's just me, and maybe a more readable and accessible command-set is in order. Ultimately, as long as knowbot can print out a help/usage page, anything should be totally fine.

**Reply by blairmunroakusa** [reply]
>So my inclination is to imitate bash. The idea being that it could help us attract developers, and also it might intrigue normal users (i.e. what are these cryptic commands?).

Ok, you and I are in the same boat. However I structure the help pages will be intuitive in either case.

**Comment by DecentralizedDan** [comment]
@blairmunroakusa This is a good question! I would do two things:
* Let's have the full set of options available as a kind of "secret menu"
  * These dev commands would intrigue users when used, as per @galactic-beyond above
* Let's abstract the non-dev options further to specific use cases
  * i.e. When would users NOT `list all`? Let's make that the default for `list`
  * Same with `search` --  the default should be `search all`

**Reply by blairmunroakusa** [reply]
> Let's have the full set of options available as a kind of "secret menu"

Yes, I like this. I plan to have one comprehensive 'menu', and then help commands for each cryptic command, eg, grep help.

> i.e. When would users NOT list all? Let's make that the default for list

Yes, the commands in either case already have that sort of default behavior. I will just need to include clear clarifications that grep can be slow and output a lot of results if you are searching _everything_ with a simple search term. ...but that is more a nuance.
